src/content/post/2023/02/21/approve-private-endpoint-connections-with-bicep.md

```diff
@@ -0,0 +1,108 @@
+tags: ["Azure", "AKS", "Kubernetes", "Private Link", "Private Endpoint"]
+series: ["Private Link"]
+---
+
+Managing private endpoint connections - especially across Azure AD tenants - can be a chore. In many cases you will have to do this manually and often you will need elevated permissions. But usually your deployment pipeline already has sufficient permissions on the resource to approve the endpoint connection, so it would be much more convenient if we could make the approval a part of our desired state config.
+
+### Approve a private endpoint with Bicep
+
+It is actually very simple to approve a private endpoint using Bicep because the private endpoint connection - which is the link between the private endpoint and the target resource - is a child resource of the target resource. 
+
+This means that if we have an Azure SQL Server that looks like this:
+
+```
+resource dbserver 'Microsoft.Sql/servers@2022-05-01-preview' = {
+  name: 'broersma'
+  location: 'westeurope'
+  properties: {
+    publicNetworkAccess: 'Disabled'
+  }
+}
+```
+
+And we have a pending private endpoint connection:
+
+![Pending Azure SQL Private Endpoint Connection](/content/approve-private-endpoint-connections-with-bicep/pending-pe-azuresql.png)
+
+We can now export the Azure SQL Server template and see that the private endpoint connection is a child resource of the Azure SQL Server!
+```json
+{
+    "type": "Microsoft.Sql/servers/privateEndpointConnections",
+    "apiVersion": "2022-05-01-preview",
+    "name": "broersma/pe-azuresql-broersma-5c1e262d-42af-40f8-8b1b-e766be1ea324",
+    "properties": {
+        "privateLinkServiceConnectionState": {
+            "status": "Pending",
+            "description": "Please approve"
+        }
+    }
+}
+```
+
+We can approve the endpoint like this
+
+```json
+{
+    "type": "Microsoft.Sql/servers/privateEndpointConnections",
+    "apiVersion": "2022-05-01-preview",
+    "name": "broersma/pe-azuresql-broersma-5c1e262d-42af-40f8-8b1b-e766be1ea324",
+    "properties": {
+        "privateLinkServiceConnectionState": {
+            "status": "Approved",
+            "description": "Approved by pipeline"
+        }
+    }
+}
+```
+
+Or as Bicep
+
+```
+resource privateEndpointConnection 'Microsoft.Sql/servers/privateEndpointConnections@2022-05-01-preview' = {
+    name: 'broersma/pe-azuresql-broersma-5c1e262d-42af-40f8-8b1b-e766be1ea324'
+    properties: {
+        privateLinkServiceConnectionState: {
+            status: 'Approved' 
+            description: 'Approved by pipeline'
+        }
+    }
+}
+```
+
+As we've seen in the previous article it is sometimes  impossible to approve the private endpoint connection through the azure portal or the azure cli.
```

Comment:
which article? the blog post?

Reply:
As we've seen in the previous post **,** it is sometimes   impossible to approve the private endpoint connection through the azure portal or the azure cli.